jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt

```diff
@@ -0,0 +1,167 @@
+import com.intellij.util.ui.UIUtil
+import org.jetbrains.yaml.YAMLFileType
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.jetbrains.services.cloudformation.Function
+import software.aws.toolkits.jetbrains.services.lambda.LambdaWidgets.lambdaMemory
+import software.aws.toolkits.jetbrains.services.lambda.LambdaWidgets.lambdaTimeout
+import software.aws.toolkits.jetbrains.services.lambda.execution.LambdaInputPanel
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamTemplateUtils.findFunctionsFromTemplate
+import software.aws.toolkits.jetbrains.services.lambda.validOrNull
+import software.aws.toolkits.jetbrains.ui.EnvironmentVariablesTextField
+import software.aws.toolkits.jetbrains.ui.HandlerPanel
+import software.aws.toolkits.jetbrains.ui.ProjectFileBrowseListener
+import software.aws.toolkits.jetbrains.ui.SliderPanel
+import software.aws.toolkits.jetbrains.utils.ui.addQuickSelect
+import software.aws.toolkits.jetbrains.utils.ui.find
+import software.aws.toolkits.resources.message
+import java.awt.event.ActionEvent
+import java.io.File
+import java.util.Comparator
+import javax.swing.DefaultComboBoxModel
+import javax.swing.JCheckBox
+import javax.swing.JComboBox
+import javax.swing.JPanel
+import javax.swing.SwingUtilities
+
+class LocalLambdaRunSettingsEditorPanel(private val project: Project) {
+    lateinit var panel: JPanel
+    lateinit var handlerPanel: HandlerPanel
+    lateinit var environmentVariables: EnvironmentVariablesTextField
+    private lateinit var runtimeModel: SortedComboBoxModel<Runtime>
+    lateinit var runtime: JComboBox<Runtime>
+    lateinit var lambdaInput: LambdaInputPanel
+    lateinit var useTemplate: JCheckBox
+    lateinit var function: JComboBox<Function>
+    private lateinit var functionModels: DefaultComboBoxModel<Function>
+    lateinit var templateFile: TextFieldWithBrowseButton
+    lateinit var lambdaInputPanel: JPanel
+    lateinit var timeoutSlider: SliderPanel
+    lateinit var memorySlider: SliderPanel
+    lateinit var invalidator: JCheckBox
+
+    var lastSelectedRuntime: Runtime? = null
+
+    private fun createUIComponents() {
+        handlerPanel = HandlerPanel(project)
+        lambdaInput = LambdaInputPanel(project)
+        functionModels = DefaultComboBoxModel()
+        function = ComboBox(functionModels)
+        function.addActionListener { updateComponents() }
+        runtimeModel = SortedComboBoxModel(compareBy(Comparator.naturalOrder()) { it: Runtime -> it.toString() })
+        runtime = ComboBox(runtimeModel)
+        environmentVariables = EnvironmentVariablesTextField()
+        timeoutSlider = lambdaTimeout()
+        memorySlider = lambdaMemory()
+    }
+
+    init {
+        lambdaInputPanel.border = IdeBorderFactory.createTitledBorder(message("lambda.input.label"), false, JBUI.emptyInsets())
+        useTemplate.addActionListener { e: ActionEvent? -> updateComponents() }
```

Comment:
Can remove e

Reply:
Removed the one below, missed this one. Removed